tests/integration/components/mirage-handler-test.js

```diff
@@ -0,0 +1,42 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { upload as uploadHandler } from 'ember-file-upload/mirage';
+import { upload } from 'ember-file-upload/test-support';
+import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
+
+module('Integration | Component | mirage-handler', function(hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  test('upload handler passed schema and request.params throw', async function(assert) {
+    let self = this;
+    this.server.post('/folder/:id/file', uploadHandler(function(schema, request) {
+      assert.step('mirage-handler');
+
+      assert.equal(schema, self.server.schema, 'schema is provided');
+      assert.ok(typeof request.params === 'object', 'params property on request is an object');
+      assert.equal(request.params.id, '1', 'value of dynamic segment is present on params object');
```

Comment:
Can you add an assertion for the requestBody properties?

```js
assert.deepEqual(request.requestBody, {
  'Content-Type': 'text/plain',
  file: {
    extension: 'txt',
    name: 'text.txt',
    size: 9,
    type: 'text/plain',
    url: 'data:text/plain;base64,c29tZS1kYXRh'
  }
});

```

Reply:
Done in a1aae2dfb41b3626c631cb0d640956b4ee22b394.